### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by Merciful12**
https://github.com/sicpa-dlab/didcomm-gemini/issues/35

### CODE REVIEW

File: site/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+const path = require('path')
+
+module.exports.onCreateNode = ({ node, actions, getNode }) => {
```
Comment (asmarty) [inline site/gatsby-node.js:5]: Please, leave TODO separate to different files

File: site/src/common/hooks/usePagination.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react'
+import { NumberParam, useQueryParam } from 'use-query-params'
+import { ITEMS_PER_PAGE } from '../constants'
+
+
+export const usePagination = <TItem extends {}>(list: Array<TItem>): {
```
Comment (asmarty) [inline site/src/common/hooks/usePagination.ts]: Can you create interface Pagination(hasNext, next,  prev, items, pagesCount, setPage) ?

File: site/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import * as React from 'react'
+import { graphql, Link, useStaticQuery } from 'gatsby'
+
+import * as styles from './Header.module.css'
+import { PropsWithChildren } from 'react'
+
+
+type QueryData = {
```
Comment (asmarty) [inline site/src/components/Header/Header.tsx:8]: Can we create file for each type? or move types to different file

**Review by asmarty — CHANGES_REQUESTED** [review]

File: site/src/components/Header/Header.tsx (hunk 3 above)
Comment (Merciful12) [reply site/src/components/Header/Header.tsx:8]: will be done in #3

**Review by asmarty — APPROVED** [review]

**Pull request merged**